nncf/torch/dynamic_graph/patch_pytorch.py

```diff
@@ -118,22 +164,39 @@ def patch_torch_operators():
     _OPERATORS_ALREADY_WRAPPED = True
 
     # patch operators
-    import torch.nn.functional as F
     import torch
-    from nncf.torch.graph.operator_metatypes import get_operator_metatypes
-    for op_meta_class in get_operator_metatypes():  # type: OperatorMetatype
-        if op_meta_class.torch_nn_functional_patch_spec is not None:
-            ps = op_meta_class.torch_nn_functional_patch_spec
-            patch_namespace_by_patchspec(F, ps)
-        if op_meta_class.torch_module_patch_spec is not None:
-            ps = op_meta_class.torch_module_patch_spec
-            patch_namespace_by_patchspec(torch, ps)
-        if op_meta_class.torch_tensor_patch_spec is not None:
-            ps = op_meta_class.torch_tensor_patch_spec
-            patch_namespace_by_patchspec(TracedTensor, ps)
+    # pylint: disable=protected-access
+    functions_to_patch = {
+        torch.nn.functional: get_all_functions_from_namespace(torch.nn.functional),
+        torch: get_all_functions_from_namespace(torch._C._VariableFunctions),
+        TracedTensor: get_all_functions_from_namespace(torch.Tensor)
+    }
+    # pylint: enable=protected-access
+    ignored_functions = IgnoredFunctions.IGNORED_FUNCTIONS
+
+    for namespace, function_names in functions_to_patch.items():
+        new_function_names = []
+        for function_name in function_names:
+            if function_name in ignored_functions:
+                continue
+            new_function_names.append(function_name)
+        functions_to_patch[namespace] = new_function_names
+
+    # Relu function from torch.nn.functional uses torch.relu or torch.relu_ inside,
+    # which we've already wrapped. So we don't have to wrap Relu in torch.nn.functional
+    # to be able to differ what torch.relu or torch.relu was called exactly inside Relu
+
+    functions_to_patch[torch.nn.functional].remove('relu')
+
+    for namespace, function_names in MagicFunctionsToPatch.MAGIC_FUNCTIONS_TO_PATCH.items():
+        functions_to_patch[namespace] += function_names
+
+    for namespace, function_names in functions_to_patch.items():
+        for function_name in function_names:
+            patch_namespace_opname(namespace, function_name)
 
     # Patch __repr__ methods so that debugging does not add new nodes to the graph
-    patch_namespace_opname(TracedTensor, PatchedOperatorInfo("__repr__", ForwardTraceOnly()))
+    # patch_namespace_opname(TracedTensor, PatchedOperatorInfo("__repr__", ForwardTraceOnly()))
```

Comment:
Commented code. We need to patch `__repr__` in a special way still so that debugging does not inflate the traced graph. BTW it would be good to add a test for this - the test that checks that if tensor contents are printed during tracing (via `print(tnsr)` or `str(tnsr)`), then no new nodes are added to the graph.

Reply:
Why should we trace this operation `__repr__`?

Reply:
Done